worldedit-core/src/main/java/com/sk89q/worldedit/command/SelectionCommands.java

```diff
@@ -504,6 +505,133 @@ private BlockVector3[] getChangesForEachDir(int amount, boolean onlyHorizontal,
         return changes.build().map(v -> v.multiply(amount)).toArray(BlockVector3[]::new);
     }
 
+    @Command(
+        name = "/trim",
+        desc = "Minimize the selection to encompass matching blocks"
+    )
+    @Logging(REGION)
+    @CommandPermissions("worldedit.selection.trim")
+    public void trim(Actor actor, World world, LocalSession session,
+                     @Arg(desc = "Mask of blocks to keep within the selection", def = "#existing")
+                         Mask mask) throws WorldEditException {
+        // Avoid checking blocks outside the original region but within the cuboid region
+        Region originalRegion = session.getSelection(world);
+        boolean skipContains = originalRegion instanceof CuboidRegion;
```

Comment:
Technically instead of doing this skipContains check in all of the below loops, you can conditionally wrap the mask with a RegionMask if it's not a cuboid region. This would bake it into the mask & simplify the code a fair bit